### PR TITLE
Drop support for EOL Python 2.7

### PR DESCRIPTION
**Pull request by hugovk**
For https://github.com/pybluez/pybluez/issues/266.

Also add `python_requires` to setup.py to help pip.

### CODE REVIEW

**Comment by hugovk** [comment]
Passing CI build:

* https://github.com/hugovk/pybluez/commit/c3d16c101a3391da40ece9b52951444a3d7920c7/checks?check_suite_id=379576864

File: bluetooth/bluez.py [hunk 1]
```diff
@@ -669,12 +669,12 @@ def device_discovered (self, address, device_class, rssi, name):
         [1] https://www.bluetooth.org/foundry/assignnumb/document/baseband
         """
         if name:
-            print(("found: %s - %s (class 0x%X, rssi %s)" % \
-                    (address, name, device_class, rssi)))
+            print("found: %s - %s (class 0x%X, rssi %s)" % \
```
Comment (cclauss) [inline bluetooth/bluez.py]: Backslash line termination is neither required inside parens nor desired according to PEP8.   These instances can be found via __flake8 . —select=E502__

Comment (hugovk) [reply bluetooth/bluez.py]: There's quite a few non-failing flake8 warnings on the CI, I'd suggest fixing those in another PR.

File: bluetooth/bluez.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 from bluetooth._bluetooth import HCI, RFCOMM, L2CAP, SCO, SOL_L2CAP, \
                                     SOL_RFCOMM, L2CAP_OPTIONS
 
-get_byte = ord if sys.version_info.major < 3 else int
+get_byte = int
```
Comment (rgov) [inline bluetooth/bluez.py:12]: Unless this is something that users are importing from this module, I'd be happy to remove uses of `get_byte` altogether.

File: bluetooth/bluez.py [hunk 3]
```diff
@@ -669,12 +669,12 @@ def device_discovered (self, address, device_class, rssi, name):
...
-            print(("found: %s - %s (class 0x%X, rssi %s)" % \
-                    (address, name, device_class, rssi)))
+            print("found: %s - %s (class 0x%X, rssi %s)" % \
+                    (address, name, device_class, rssi))
         else:
-            print(("found: %s (class 0x%X)" % (address, device_class)))
-            print(("found: %s (class 0x%X, rssi %s)" % \
-                    (address, device_class, rssi)))
+            print("found: {} (class 0x{:X})".format(address, device_class))
+            print("found: %s (class 0x%X, rssi %s)" % \
```
Comment (rgov) [inline bluetooth/bluez.py]: Could you change this to use `.format` as well for consistency?

**Review by rgov — CHANGES_REQUESTED** [review]
Thanks for this! I'd like to see this exercised on different platforms before we release it, but it looks pretty good.

File: bluetooth/bluez.py (hunk 2 above)
Comment (hugovk) [reply bluetooth/bluez.py:12]: Any idea how to find out if people are using it?

Comment (rgov) [reply bluetooth/bluez.py:12]: Unfortunately it's used in one of the examples: https://github.com/pybluez/pybluez/blob/master/examples/advanced/inquiry-with-rssi.py#L102

Python doesn't give us a great way to deprecate it. My preference would just be to remove it altogether and make a note in the CHANGELOG that it is no longer necessary.

File: bluetooth/bluez.py (hunk 3 above)
Comment (hugovk) [reply bluetooth/bluez.py]: Done!

File: bluetooth/bluez.py (hunk 2 above)
Comment (rgov) [reply bluetooth/bluez.py:12]: Filed #299.

**Review by rgov — APPROVED** [review]

**Pull request merged**